allocator/mimalloc-sys/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "mimalloc-sys"
+build = "build.rs"
+links = "mimalloc"
+authors.workspace = true
+homepage.workspace = true
+repository.workspace = true
+edition.workspace = true
+license.workspace = true
+version.workspace = true
+
+[dependencies]
+
+[lints]
+workspace = true
+
+[build-dependencies]
+cmake = "0.1"
+cc = "1.0"
```

Comment:
Is it possible to put all of this behind a feature flag? The new dependency is a bit "ouch". I love the idea of omitting it for local dev builds but using it in CI & release builds.

Reply:
If it adds 2.5 seconds to build time, I'd rather we dogfood it and it be in dev builds by default, rather than all our local testing NOT match what we build/ship and then being surprised by something when we do.

Reply:
Either way, a feature flag approach would at least clean up the checks that look like `[target.'cfg(not(any(target_family = "wasm")))'.dependencies]` . (It could be on by default but wasm can opt out).

Reply:
@billti @minestarks There is the question of whether we have this set in all libs. Right now, it is only set on distribution units, so any tests in those projects will use mimalloc. Any low level libs, for their individual crate tests, would still be using the default allocator. We can change this, but we have the same issue with WASM. Don't run all of the tests compiled to wasm, just the integration tests.